### PR TITLE
[LF] cleanup transaction doc

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: daml-lf/spec/transaction.rst [hunk 1]
```diff
@@ -246,15 +246,16 @@ An action on the ledger.
...
 * ``string`` `node_id`
 
-``version``  is optional. If unset it should be interpreted as version 10.
-otherwise it should be constraint as described in `field version`_
+Field ``version`` is required and must be a valid version as described under `field
+version`_, not newer that the version of the enclosing Transaction
+message.
 
-``node_id`` is required. it is csontraint as described under `field
+``node_id`` is required. it is constraint as described under `field
```
Comment (paulbrauner-da) [inline daml-lf/spec/transaction.rst:258]: Should it be "constrained"?

File: daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto [hunk 2]
```diff
@@ -49,26 +44,26 @@ message Node {
...
 
-    string version = 63; // *since 11*, optional
+    string version = 63;
 }
 
 message KeyWithMaintainers {
     oneof key {
-        com.daml.lf.value.VersionedValue key_versioned = 1; // *until version 11*
-        bytes key_unversioned = 3; // *defined in version >= 1.11*
+        com.daml.lf.value.VersionedValue key_versioned = 1; // *must not be used*
```
Comment (paulbrauner-da) [inline daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:55]: Protobufs have a way to mark a field as deprecated: `[deprecated = true]` (see https://protobuf.dev/programming-guides/proto3/#options). Should we use this here?

**Review by paulbrauner-da — APPROVED** [review]
LGTM, thanks!

File: daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:55]: Let start with that in daml 3.0

File: daml-lf/spec/transaction.rst (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/spec/transaction.rst:258]:
```suggestion
``node_id`` is required. it is constrained as described under `field
```

**Pull request merged**